web/src/narrow.js

```diff
@@ -207,7 +247,14 @@ export function activate(raw_terms, opts) {
     if (raw_terms.length === 0) {
         raw_terms = [{operator: "is", operand: "home"}];
     }
-    const filter = new Filter(raw_terms);
+
+    // raw_terms with both `dm` and with operators are redundant, since dms
+    // don't have topics. Hence, we remove the `with` operator.
+    if (raw_terms.some((term) => term.operator === "dm")) {
+        raw_terms = raw_terms.filter((term) => term.operator !== "with");
+    }
```

Comment:
I assume private messages won't require the `with` operator.

Reply:
I think that's correct, it should be a noop on DMs for now. 

But we may change that behavior. I'll link to https://github.com/zulip/zulip/issues/20501 so that there's a reference on that issue of this detail.